actions/workflows/create_vm_role.yaml

```diff
@@ -43,13 +41,13 @@
         keyfile: "{{keyfile}}"
         ssh_timeout: 20
         retries: 30
-      on-success: "add_name_tag"
+      on-success: "add_tags"
     -
-      name: "add_name_tag"
+      name: "add_tags"
       ref: "aws.ec2_create_tags"
       params:
         resource_ids: "{{ec2_instance_id}}"
-        tags: "Name={{hostname}}"
+        tags: "Name={{hostname}},Creator={{creator}},Role={{role}}{% if purpose %},Purpose={{purpose[0]|upper}}{{purpose[1:]}}{% endif %}"
```

Comment:
Title capitalizes every word :( There's also "capitalize", but it puts everything else in lowercase. So had to go with what you see now :)